src/main/scala/org/allenai/ike/IkeKryoRegistrator.scala

```diff
@@ -0,0 +1,53 @@
+package org.allenai.ike
+
+import org.allenai.ike.patterns.NamedPattern
+
+import com.esotericsoftware.kryo.io.{ Input, Output }
+import com.esotericsoftware.kryo.{ Kryo, Serializer }
+import org.apache.spark.serializer.KryoRegistrator
+
+/** Helper to register custom serializers for Some and None. Since Kryo is a Java library, it
+  * doesn't provide default serializers for these classes.
```

Comment:
That's not how I would make that trade-off, but it's a reasonable reason.

I would go with Chill on autopilot. On reflection though, the reason is that otherwise, when you develop this more, you end up with situations where it works with a 2-tuple, but then you change the code a bit to use a 3-tuple, and suddenly it stops working at runtime, in a part of the code you know nothing about.

Like I said though, no big deal. Keeping the dependencies light has its own perks.